blocks/editable/index.js

```diff
@@ -123,6 +126,38 @@ export default class Editable extends Component {
 		}
 	}
 
+	onPastePostProcess( event ) {
+		const childNodes = Array.from( event.node.childNodes );
+		const isBlockDelimiter = ( node ) =>
+			node.nodeType === 8 && /^ wp:/.test( node.nodeValue );
+		const isDoubleBR = ( node ) =>
+			node.nodeName === 'BR' && node.previousSibling && node.previousSibling.nodeName === 'BR';
+		const isBlockPart = ( node ) =>
+			isDoubleBR( node ) || this.editor.dom.isBlock( node );
+
+		// If there's no `onSplit` prop, content will later be converted to
+		// inline content.
+		if ( this.props.onSplit ) {
+			let blocks = [];
+
+			// Internal paste, so parse.
+			if ( childNodes.some( isBlockDelimiter ) ) {
+				blocks = parse( event.node.innerHTML.replace( /<meta[^>]+>/, '' ) );
+			// External paste with block level content, so attempt to assign
+			// blocks.
+			} else if ( childNodes.some( isBlockPart ) ) {
+				blocks = pasteHandler( childNodes );
+			}
```

Comment:
Maybe we could split out the logic here out of the `Editable` into the `parse.js` file. I mean the `pasteHandler` could probably check for the block delimiters itself, don't you think?